src/T3.java

```diff
@@ -0,0 +1,87 @@
+public class T3 {
+    private Player[][] board = new Player[3][3];
+
+    public enum Player {
+        X, O, NONE
+    }
+    private Player currentPlayer = Player.X;//There has to be a player whose turn it is
+
+    public Player getPosition(int row, int col) {
+        if(row > 2 || row < 0 || col > 2 || col < 0){
+            throw new IndexOutOfBoundsException();
+        }
+        return board[row][col];
+    }
+
+    public void makeMove(int row, int col) throws IllegalArgumentException {
+        if(getPosition(row, col) == Player.X || getPosition(row, col) == Player.O){
```

Comment:
`getPosition(row, col) != Player.__` is simpler...same for more detail which player is occupying